internal/api/v1/handlers/instance.go

```diff
@@ -156,3 +156,159 @@ func (h *InstanceHandler) GetInstance(c *fiber.Ctx) error {
+
+	// Get pagination parameters
+	page := c.QueryInt("page", 1)
+	if page < 1 {
+		page = 1
+	}
+	limit := c.QueryInt("limit", 10)
+	if limit < 1 {
+		limit = 10
+	}
+	offset := (page - 1) * limit
+
+	// Get instances with their details using the service
+	instances, err := h.service.GetPublicIPs(c.Context(), &models.ListOptions{
+		Limit:  limit,
+		Offset: offset,
+	})
+	if err != nil {
+		fmt.Printf("❌ Error getting instance metadata: %v\n", err)
+		return c.Status(fiber.StatusInternalServerError).JSON(fiber.Map{
+			"error": fmt.Sprintf("failed to get instance metadata: %v", err),
+		})
+	}
+
+	fmt.Printf("✅ Found %d instances\n", len(instances))
+
+	// Convert instances to the format expected by the test
+	result := make([]map[string]interface{}, len(instances))
+	for i, instance := range instances {
+		result[i] = map[string]interface{}{
+			"id":         instance.ID,
+			"job_id":     instance.JobID,
+			"public_ip":  instance.PublicIP,
+			"name":       instance.Name,
+			"region":     instance.Region,
+			"size":       instance.Size,
+			"image":      instance.Image,
+			"tags":       instance.Tags,
+			"status":     instance.Status,
+			"created_at": instance.CreatedAt,
+		}
+	}
+
+	// Return instances with pagination info
+	return c.JSON(fiber.Map{
+		"instances": result,
+		"total":     len(instances),
+		"page":      page,
+		"limit":     limit,
+		"offset":    offset,
+	})
+}
+
+// GetInstancesByJobID returns a list of instances for a specific job
+func (h *InstanceHandler) GetInstancesByJobID(c *fiber.Ctx) error {
+	jobID, err := c.ParamsInt("jobId")
+	if err != nil {
+		return c.Status(fiber.StatusBadRequest).JSON(fiber.Map{
+			"error": "invalid job id",
+		})
+	}
+	if jobID <= 0 {
+		return c.Status(fiber.StatusBadRequest).JSON(fiber.Map{
+			"error": "job id is required",
+		})
+	}
+
+	fmt.Printf("🔍 Getting instances for job ID %d...\n", jobID)
+
+	// Get instances using the service
+	instances, err := h.service.GetInstancesByJobID(c.Context(), uint(jobID))
+	if err != nil {
+		fmt.Printf("❌ Error getting instances for job %d: %v\n", jobID, err)
+		return c.Status(fiber.StatusInternalServerError).JSON(fiber.Map{
+			"error": fmt.Sprintf("failed to get instances for job %d: %v", jobID, err),
+		})
+	}
+
+	fmt.Printf("✅ Found %d instances for job %d\n", len(instances), jobID)
+
+	// Convert instances to the format expected by the test
+	result := make([]map[string]interface{}, len(instances))
+	for i, instance := range instances {
+		result[i] = map[string]interface{}{
+			"id":         instance.ID,
+			"job_id":     instance.JobID,
+			"public_ip":  instance.PublicIP,
+			"name":       instance.Name,
+			"region":     instance.Region,
+			"size":       instance.Size,
+			"image":      instance.Image,
+			"tags":       instance.Tags,
+			"status":     instance.Status,
+			"created_at": instance.CreatedAt,
+		}
+	}
+
+	// Return all instance details
+	return c.JSON(fiber.Map{
+		"instances": result,
```

Comment:
You should be able to simply write `instances` here instead of `result` and all fields should come. No need to iterate over them and convert them to map.